lib/file-operations.js

```diff
@@ -0,0 +1,91 @@
+const fs = require("fs");
+const { access, constants } = require ('node:fs/promises');
+function makeDir(folder){
+return new Promise((resolve,reject)=>{
+    fs.mkdir(folder,(err)=>{
+        if(err){
+            resolve({success: false,error : err})
+        }
+        else {
+            resolve({success: true })
+        }
+    })
+})
+}
+
+function deleteFile(file){
+    return new Promise((resolve, reject)=>{
+        fs.unlink(file,(err)=>{
+            if(err){
+                resolve({success:false,error : err})
+            }
+            else {
+                resolve({success : true})
+            }
+        })
+    })
+}
+
+function copyFile(source,dest){
+    return new Promise((resolve)=>{
+        fs.copyFile(source,dest,(err)=>{
+            if(err){
+                resolve({success : false,error:err})
+            }
+            else {
+                resolve({success : true})
+            }
+        })
+    })
+}
+
+function moveFile(source,dest){
+    return new Promise(resolve => {
+        fs.rename(source,dest,(err)=>{
+            if(err){
+                resolve({success : false, error : err})
+            }
+            else {
+                resolve({success : true})
+            }
+        })
+    })
+}
+
+function deleteFolder(path){
+    return new Promise(resolve => {
+        fs.rmdir(path,(err)=>{
+            if (err){
+                resolve({success : false, error : err})
+            }
+            else {
+                resolve({success : true})
```

Comment:
now that we have error enums, using that to resolve might make the code more readable